drivers/gpio/gpio_sam4l.c

```diff
@@ -0,0 +1,305 @@
+		gpio->GPERS = mask;
+		/* Disable output. */
+		gpio->ODERC = mask;
+
+		return 0;
+	}
+
+	/* Setup the pin direcion. */
+	if (flags & GPIO_OUTPUT) {
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			/* Set the pin. */
+			gpio->OVRS = mask;
+		}
+		if (flags & GPIO_OUTPUT_INIT_LOW) {
+			/* Clear the pin. */
+			gpio->OVRC = mask;
+		}
+		/* Enable the output */
+		gpio->ODERS = mask;
+		gpio->STERC = mask;
+	} else {
+		/* Disable the output */
+		gpio->ODERC = mask;
+		gpio->STERS = mask;
+	}
+
+	/* Note: Input is always enabled. */
+
+	/* Setup selected Pull resistor.
+	 *
+	 * A pull cannot be enabled if the opposite pull is enabled.
+	 * Clear both pulls, then enable the one we need.
+	 */
+	gpio->PUERC = mask;
+	gpio->PDERC = mask;
+	if (flags & GPIO_PULL_UP) {
+		/* Enable pull-up. */
+		gpio->PUERS = mask;
+	/* Setup Pull-down resistor. */
+	} else if (flags & GPIO_PULL_DOWN) {
+		/* Enable pull-down. */
+		gpio->PDERS = mask;
+	}
+
+	/* Enable the GPIO to control the pin (instead of a peripheral). */
+	gpio->GPERS = mask;
+
+	return 0;
+}
+
+static int gpio_sam_config(const struct device *dev,
+			   gpio_pin_t pin,
+			   gpio_flags_t flags)
+{
+	return gpio_sam_port_configure(dev, BIT(pin), flags);
+}
+
+static int gpio_sam_port_get_raw(const struct device *dev,
+				 uint32_t *value)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Gpio * const gpio = cfg->regs;
+
+	*value = gpio->PVR;
+
+	return 0;
+}
+
+static int gpio_sam_port_set_masked_raw(const struct device *dev,
+					uint32_t mask,
+					uint32_t value)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Gpio * const gpio = cfg->regs;
+
+	gpio->OVR = (gpio->PVR & ~mask) | (mask & value);
+
+	return 0;
+}
+
+static int gpio_sam_port_set_bits_raw(const struct device *dev,
+				      uint32_t mask)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Gpio * const gpio = cfg->regs;
+
+	/* Set pins. */
+	gpio->OVRS = mask;
+
+	return 0;
+}
+
+static int gpio_sam_port_clear_bits_raw(const struct device *dev,
+					uint32_t mask)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Gpio * const gpio = cfg->regs;
+
+	/* Clear pins. */
+	gpio->OVRC = mask;
+
+	return 0;
+}
+
+static int gpio_sam_port_toggle_bits(const struct device *dev,
+				     uint32_t mask)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Gpio * const gpio = cfg->regs;
+
+	/* Toggle pins. */
+	gpio->OVRT = mask;
+
+	return 0;
+}
+
+static int gpio_sam_port_interrupt_configure(const struct device *dev,
+					     uint32_t mask,
+					     enum gpio_int_mode mode,
+					     enum gpio_int_trig trig)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Gpio * const gpio = cfg->regs;
+
+	/* Disable the interrupt. */
+	gpio->IERC = mask;
+	/* Set default to both edges */
+	gpio->IMR0C = mask;
+	gpio->IMR1C = mask;
+
+	if (trig != GPIO_INT_TRIG_BOTH) {
+		if (trig == GPIO_INT_TRIG_HIGH) {
+			gpio->IMR0S = mask;
+			gpio->IMR1C = mask;
```

Comment:
`gpio->IMR1C` was cleared above in `/* Set default to both edges */`, we don't need to do it again.

Reply:
Fixed!